Makefile

```diff
@@ -366,15 +366,17 @@ terratest/install: | guard/program/go
 	cd $(TERRAFORM_TEST_DIR) && go mod tidy
 	@ echo "[$@]: Completed successfully!"
 
+terratest/test: TERRATEST_FILES ?= $(shell find $(TERRAFORM_TEST_DIR) -name "*.go")
 terratest/test: | guard/program/go
 terratest/test: TIMEOUT ?= 20m
 terratest/test:
-	@ echo "[$@] Starting Terraform tests"
-	cd $(TERRAFORM_TEST_DIR) && go test -count=1 -timeout $(TIMEOUT)
+	@ echo "[$@] Starting Terratest-based Terraform tests"
+	$(if $(TERRATEST_FILES),\
```

Comment:
I'm a little hesitant on this kind of change... It's very close to a false positive... i.e. the user has told us to run `terratest/test`, and instead of erroring or alerting because there aren't any tests, we're claiming success with a zero exit-code. This seems counter-intuitive to me. As a user, I _want_ an indicator when a command I am issuing is incorrect or invalid, typically with a non-zero exit so I can gate my own logic on it.

Reply:
I knew this PR was going to generate discussion because I don't think testing has been very thought through:

- `make test` is used for installation tests and not unit tests.  There's no `make` target for unit tests.
-  My understanding is that `make test` is invoked when a special line appears in the PR because an AWS account ID is needed for the tests and that's not handled well by Travis.  So you would know what you're doing and have go files when you invoke this command.
- Even if you could get around the issues with Travis, you couldn't standardize on running `make test` similar to `make lint` because when there are no Terratests the `make tests` would fail.

I'm trying to work around `make tests` and find a way to add my unit tests.  Whatever I propose, you will override, so tell me how you want to handle unit and installation tests.  The solution would have to consider those tests that require an AWS account ID.